internal/connection/spinner.go

```diff
@@ -0,0 +1,94 @@
+package connection
+
+import (
+	"fmt"
+
+	"github.com/charmbracelet/bubbles/spinner"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+)
+
+const spinnerText = `Connecting to the cluster %s at %s.
+   Check the logs at %s.`
+
+var (
+	currSpinner  = spinner.Dot
+	spinnerStyle = lipgloss.NewStyle().Foreground(lipgloss.Color("#40826D"))
+	helpStyle    = lipgloss.NewStyle().Foreground(lipgloss.Color("241")).Render
+)
+
+type connectionSpinnerModel struct {
+	clusterName string
+	address     string
+	logFileName string
+	spinner     spinner.Model
+	escaped     *bool
```

Comment:
Why is this a pointer to bool?

Reply:
The address is given outside of the TUI program and the content is being changed by the TUI to determine whether user escaped.